modules/tinymce/src/core/test/ts/webdriver/paste/CopyAndPasteTest.ts

```diff
@@ -0,0 +1,124 @@
+import { RealClipboard, RealMouse } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyHooks, TinySelections, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('webdriver.tinymce.core.paste.CopyAndPasteTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    toolbar: false,
+    statusbar: false
+  }, []);
+
+  type Selection = [ number[], number, number[], number ];
```

Comment:
Maybe use the CursorPath type from agar Cursors here. It is an object structure though so `{ startPath: [0], soffset: 0, finishPath: [0], foffset: 0 }` having an object is a bit easier to read I think.

Reply:
ok, actually, I used this signature to imitate functions like `setSelection`, changing it to an object would result in this:

```
await pCopyAndPaste(
  editor,
  { startPath: [ 0, 0 ], soffset: 0, finishPath: [ 0, 0 ], foffset: 3 },
  { startPath: [ 2, 0 ], soffset: 1, finishPath: [ 2, 0 ], foffset: 3 }
);
```

could make sense to make a helper function like this?

```
await pCopyAndPaste(
  editor,
  toSelection([ 0, 0 ], 0, [ 0, 0 ], 3),
  toSelection([ 2, 0 ], 1, [ 2, 0 ], 3)
);
```

or

```
await pCopyAndPaste(
  editor, {
    source: toSelection([ 0, 0 ], 0, [ 0, 0 ], 3),
    target: toSelection([ 2, 0 ], 1, [ 2, 0 ], 3)
  }
);
```

in this way, the type `Selection` would be more readable and the signature of `pCopyAndPaste` would stay similar to the other functions like `setSelection`

Reply:
The problem with both of those aren't as readable... the named approach lets you see what each component signifies with no background required for what the type/order is.

Reply:
ok, so I'll go with the object solution, thanks

Reply:
done